src/yellowhammer/llm.py

```diff
@@ -25,6 +25,26 @@ class FinalResponse(BaseModel):
     final_output: Union[code, ConversationalResponse]
 
 
+def error_parser(output):
+    """
+    Parse the API output to handle errors gracefully.
+    """
+    if output["parsing_error"]:
+        raw_output = str(output["raw"].content)
+        error = output["parsing_error"]
+        out_string = f"Error parsing LLM output. Parse error: {error}. \n Raw output: {raw_output}"
+        return FinalResponse(final_output=ConversationalResponse(response=out_string))
+
+    elif not output["parsed"]:
+        raw_output = str(output["raw"].content)
+        out_string = f"Error in LLM response. \n Raw output: {raw_output}"
+        return FinalResponse(final_output=ConversationalResponse(response=out_string))
+
```

Comment:
I guess this is naturally where linting could be built into this magic? (c.f. #9) Does the raw output here always consist of code, or does it include any preamble from the agent?

Reply:
Probably linting can come after the error parsing - we could put another link at the end of the chain. This is just to handle LLM failure modes. Can put in logic so that "if code output, apply linter, return code"

The raw output is a dict that should contain either a `parsed` entry containing the pydantic output, or a `parsing error` entry.

If `parsed` exists, it can choose from pure text or the `code` pydantic class which contains three blocks - a text prefix, imports code and body code.